chain/ethereum.ts

```diff
@@ -345,67 +327,67 @@ export namespace ethereum {
    * An Ethereum block.
    */
   export class Block {
-    hash: Bytes
-    parentHash: Bytes
-    unclesHash: Bytes
-    author: Address
-    stateRoot: Bytes
-    transactionsRoot: Bytes
-    receiptsRoot: Bytes
-    number: BigInt
-    gasUsed: BigInt
-    gasLimit: BigInt
-    timestamp: BigInt
-    difficulty: BigInt
-    totalDifficulty: BigInt
-    size: BigInt | null
+    hash!: Bytes
+    parentHash!: Bytes
+    unclesHash!: Bytes
+    author!: Address
+    stateRoot!: Bytes
+    transactionsRoot!: Bytes
+    receiptsRoot!: Bytes
+    number!: BigInt
+    gasUsed!: BigInt
+    gasLimit!: BigInt
+    timestamp!: BigInt
+    difficulty!: BigInt
+    totalDifficulty!: BigInt
+    size!: BigInt | null
```

Comment:
hmm what's the meaning of `!` in a field that is `| null`? I see this happens a few times.

Reply:
It's because of [this](https://github.com/graphprotocol/graph-docs/pull/83/files#diff-fc7d8a215aaf99e3614048e98c5db52e125742d5f146d96623a45c53f4f255eaR296).

Basically when the properties of an exported class don't have a constructor, it requires to either create one, or to add the `!`. I think its related to how you intend on setting them.

I prefered this way instead of the constructors because as far as I know only `graph-node` will populate these and pass to the subgraph developer.

Reply:
Note that annotating with `!` will yield a runtime check in Wasm, while assigning in the constructor will not.

Reply:
Thanks for the warning! I guess I'll change it then 🙂

Reply:
@dcodeIO I was discussing here with @leoyvens and we're a bit confused.

This check you're talking about, is it like an assertion to see if the property is null or not?

Reply:
A runtime check is inserted whenever a property annotated with `!` is accessed (this is essentially the developer saying "I promise this is always set"), guaranteeing that the value is in fact always set. On the contrary, if the constructor would always assign a non-nullable value, or if the type of the field would be nullable, such a runtime check would not be necessary, as the condition can be evaluated statically.